src/pages/Recipes/hooks/useRecipesQuery.ts

```diff
@@ -0,0 +1,22 @@
+import { useQuery, type UseQueryResult } from "@tanstack/react-query";
+import { fetchAllRecipes, type FetchAllRecipesResponse } from "../api";
+import type { RecipeListing } from "../../../types";
+import camelize from "camelize-ts";
+
+type RecipesQueryResult = UseQueryResult<FetchAllRecipesResponse> & {
+  recipes?: RecipeListing[];
+};
+
+const useRecipesQuery = (): RecipesQueryResult => {
+  const query = useQuery({
+    queryKey: ["recipes"],
+    queryFn: fetchAllRecipes,
+  });
+
+  return {
+    ...query,
+    recipes: camelize(query.data?.recipes),
```

Comment:
I'm using `camelize`to convert the API response from snake case to camel case, do we have any similar tooling in the frontend repo? I couldn't see any from a quick look.